### PR TITLE
Implements update quest

### PR DESCRIPTION
**Pull request by Rotorsoft**
## Link to Issue
Closes: #9385

## Description of Changes
- Implements new command
- Fixes small issues in quest model (with migration)
- Adds unit tests

## Test Plan
- Pass units

## Deployment Plan

1. 

## Other Considerations

-

### CODE REVIEW

File: libs/model/src/quest/UpdateQuest.command.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            })),
+          );
+        }
+
+        const delta = getDelta(quest, {
+          name,
+          description,
+          start_date,
+          end_date,
+        });
```
Comment (rbennettcw) [inline libs/model/src/quest/UpdateQuest.command.ts:75]: is this an alternative to `if (x) !== 'undefined'`?

File: libs/schemas/src/entities/quest.schemas.ts [hunk 2]
```diff
@@ -32,6 +32,7 @@ export const QuestActionMeta = z
     participation_period: z.nativeEnum(QuestParticipationPeriod).optional(),
     participation_times_per_period: z.number().optional(),
     created_at: z.coerce.date().optional(),
+    updated_at: z.coerce.date().optional(),
```
Comment (rbennettcw) [inline libs/schemas/src/entities/quest.schemas.ts:35]: The migration removes the `updated_column`, but here it's added. Should it exist or no?

Comment (Rotorsoft) [reply libs/schemas/src/entities/quest.schemas.ts:35]: this is the action meta, which can potentially be updated (not sure)... but the migration removes updated_at in the actions table, that's append only

**Review by rbennettcw — APPROVED** [review]

**Review by kurtisassad — APPROVED** [review]
lgtm

**Pull request merged**